Reformatted VS Code documentation

jasonviviano opened the pull request:
Thank you for helping make the Dapr documentation better!

**Please follow this checklist before submitting:**

- [ ] [Read the contribution guide](https://docs.dapr.io/contributing/contributing-docs/)
- [ ] Commands include options for Linux, MacOS, and Windows within codetabs
- [ ] New file and folder names are globally unique
- [ ] Page references use shortcodes instead of markdown or URL links
- [ ] Images use HTML style and have alternative text
- [ ] Places where multiple code/command options are given have codetabs

In addition, please fill out the following to help reviewers understand this pull request:

## Description

Reformatted some of the information in the VS Code Dapr documentation by creating a folder instead of singular file to hold information regarding

- Extension
- Remote Dev Work
- Manually setting up VS Code debugging for Dapr

jasonviviano committed:
Reformatted VS Code documentation

ghost commented:
[![CLA assistant check](https://cla.opensource.microsoft.com/pull/badge/signed)](https://cla.opensource.microsoft.com/dapr/docs?pullRequest=1601) <br/>All CLA requirements met.

AaronCrawfis commented on daprdocs/content/en/developing-applications/ides/visual-studio-code/_index.md:
```diff
@@ -0,0 +1,7 @@
+---
+type: docs
+title: "Visual Studio Code"
```
```suggestion
title: "Visual Studio Code integration with Dapr"
```

AaronCrawfis commented on daprdocs/content/en/developing-applications/ides/visual-studio-code/_index.md:
```diff
@@ -0,0 +1,7 @@
+title: "Visual Studio Code"
+linkTitle: "Visual Studio Code"
+weight: 1000
+description: "Information on how to develop and run Dapr applications in VS Code"
```
```suggestion
description: "How to develop and run Dapr applications in Visual Studio Code"
```

AaronCrawfis commented on daprdocs/content/en/developing-applications/ides/visual-studio-code/vscode-dapr-extension.md:
```diff
@@ -0,0 +1,140 @@
+title: "Dapr Visual Studio Code extension overview"
+linkTitle: "Overview"
+weight: 10000
+description:  "Information on how to develop and run Dapr applications with the Dapr extension"
```
```suggestion
description:  "How to develop and run Dapr applications with the Dapr extension"
```

AaronCrawfis commented on daprdocs/content/en/developing-applications/ides/visual-studio-code/vscode-dapr-extension.md:
```diff
@@ -0,0 +1,140 @@
+---
+type: docs
+title: "Dapr Visual Studio Code extension overview"
+linkTitle: "Overview"
```
```suggestion
linkTitle: "Dapr extension"
```

jasonviviano replied:
Hey with this change, I believe @orizohar might prefer the Overview link title, he mentioned that making this overview would give users a place to start in.

orizohar replied:
I think it all depends on the context. When the hierarchy just said "Visual Studio Code" I can see Aaron's point. Now that you updated it to "Visual Studio Code integration with Dapr" I think "Overview" makes sense and is consistent with the rest of the docs

AaronCrawfis replied:
The issue here is that the Visual Studio Code section has 3 articles in it which are disjoint: a user won't go to both the extension page and the manual setup page. It's one or the other. Overview means all the pages are related and it's introducing all the pages. In this case we want users to pick one of the 3 options. If anything we need a separate page called "Overview", but not this one

jasonviviano replied:
I do agree that there are disjointed in the sense that they don't necessarily go hand in hand. What would a "Overview" page for this even look like though considering its a IDE page?

AaronCrawfis commented on daprdocs/content/en/developing-applications/ides/visual-studio-code/vscode-dapr-extension.md:
```diff
@@ -0,0 +1,140 @@
+
+<a href="vscode:extension/ms-azuretools.vscode-dapr" class="btn btn-primary" role="button">Open in VSCode</a>
+
+## Features Overview
```
```suggestion
## Features
```

AaronCrawfis commented on daprdocs/content/en/developing-applications/ides/visual-studio-code/vscode-dapr-extension.md:
````diff
@@ -0,0 +1,140 @@
+            "request": "launch",
+            "name": "Launch Program",
+            "skipFiles": [
+                "<node_internals>/**"
+            ],
+            "program": "${workspaceFolder}/app.js"
+        },
+        {
+            "type": "pwa-node",
+            "request": "launch",
+            "name": "Launch Program with Dapr",
+            "skipFiles": [
+                "<node_internals>/**"
+            ],
+            "program": "${workspaceFolder}/app.js",
+            "preLaunchTask": "daprd-debug",
+            "postDebugTask": "daprd-down"
+        }
+}
+```
+After scaffolding task.json:
+
+```json
+{
+	"version": "2.0.0",
+	"tasks": [
+		{
+			"appId": "nodeapp",
+			"appPort": 3500,
+			"label": "daprd-debug",
+			"type": "daprd"
+		},
+		{
+			"appId": "nodeapp",
+			"label": "daprd-down",
+			"type": "daprd-down"
+		}
+	]
+}
+```
````
```suggestion
### Scaffold Dapr debugging tasks

The Dapr extension helps you debug your applications with Dapr using Visual Studio Code's [built-in debugging capability](https://code.visualstudio.com/Docs/editor/debugging).

The `Dapr: Scaffold Dapr Tasks` [Command Palette](https://code.visualstudio.com/docs/getstarted/userinterface#_command-palette) operation will update your `task.json` and `launch.json` files to launch and configure the Dapr sidecar when you begin debugging.

1. Make sure you have a launch configuration set for your app. ([Learn more](https://code.visualstudio.com/Docs/editor/debugging))
1. Open the Command Palette with `Ctrl+Shift+P`
1. Select `Dapr: Scaffold Dapr Tasks`
1. Run your app and the Dapr sidecar with `F5` or via the Run view.
```

AaronCrawfis replied:
Still applicable

AaronCrawfis commented on daprdocs/content/en/developing-applications/ides/visual-studio-code/vscode-dapr-extension.md:
````diff
@@ -0,0 +1,140 @@
+}
+```
+
+### Scaffold Dapr Components 
+* Generates the Dapr component assets needed for general Dapr applications.
+* Provides a Component folder in your VS Code workspace that contains pubsub, statestore and a zipkin yaml file.
+* Example pubsub file:
+
+```yaml
+apiVersion: dapr.io/v1alpha1
+kind: Component
+metadata:
+  name: pubsub
+spec:
+  type: pubsub.redis
+  metadata:
+  - name: redisHost
+    value: localhost:6379
+  - name: redisPassword
+    value: ""
+```
````
```suggestion
### Scaffold Dapr application components

When adding Dapr to your application, you may want to have a dedicated components directory, separate from the default components initialized as part of `dapr init`.

To create a dedicated components folder with the default `statestore`, `pubsub`, and `zipkin` components, use the `Dapr: Scaffold Dapr Components` [Command Palette](https://code.visualstudio.com/docs/getstarted/userinterface#_command-palette) operation.

1. Open your application directory in Visual Studio Code
1. Open the Command Palette with `Ctrl+Shift+P`
1. Select `Dapr: Scaffold Dapr Components`
1. Run your application with `dapr run --components-path ./components -- ...`
```

AaronCrawfis replied:
Still applicable

AaronCrawfis commented on daprdocs/content/en/developing-applications/ides/visual-studio-code/vscode-dapr-extension.md:
````diff
@@ -0,0 +1,140 @@
+    value: ""
+```
+
+### View Running Dapr Applications
+* Adds a view container specifically for Dapr applications which when selected displays a tree view called "Applications" which shows locally-running Dapr applications.
+
+<br /><img src="/images/vscode-extension-view.png" alt="Screenshot of the Dapr VSCode extension view running applications option" width="800">
+
````
```suggestion
### View running Dapr applications

The Applications view shows Dapr applications running locally on your machine.

<br /><img src="/images/vscode-extension-view.png" alt="Screenshot of the Dapr VSCode extension view running applications option" width="800">
```

AaronCrawfis replied:
Still applicable

AaronCrawfis commented on daprdocs/content/en/developing-applications/ides/visual-studio-code/vscode-dapr-extension.md:
```diff
@@ -0,0 +1,140 @@
+
+<br /><img src="/images/vscode-extension-view.png" alt="Screenshot of the Dapr VSCode extension view running applications option" width="800">
+
+### Invoke Dapr Application Methods
+* Allows users to select a Dapr application found in the tree view "Applications" and invoke GET/POST methods by name.
+* Allows users to specify an optional payload for POST methods
+<br /><img src="/images/vscode-extension-invoke.png" alt="Screenshot of the Dapr VSCode extension invoke option" width="800">
+
```
```suggestion
### Invoke Dapr applications

Within the Applications view, users can right-click and invoke Dapr apps via GET or POST methods, optionally specifying a payload.

<br /><img src="/images/vscode-extension-invoke.png" alt="Screenshot of the Dapr VSCode extension invoke option" width="800">
```

AaronCrawfis replied:
Still applicable

AaronCrawfis commented on daprdocs/content/en/developing-applications/ides/visual-studio-code/vscode-dapr-extension.md:
```diff
@@ -0,0 +1,140 @@
+* Allows users to specify an optional payload for POST methods
+<br /><img src="/images/vscode-extension-invoke.png" alt="Screenshot of the Dapr VSCode extension invoke option" width="800">
+
+### Publish Events to Dapr Applications
+* You can publish events to running applications you select in the "Applications" tree view.
+* Allows you to specify both the topic and an optional payload.
+
+  <br /><img src="/images/vscode-extension-publish.png" alt="Screenshot of the Dapr VSCode extension publish option" width="800">
+
+
```
```suggestion
### Publish events to Dapr applications

Within the Applications view, users can right-click and publish messages to a running Dapr application, specifying the topic and payload.

Users can also publish messages to all running applications.

  <br /><img src="/images/vscode-extension-publish.png" alt="Screenshot of the Dapr VSCode extension publish option" width="800">
```

AaronCrawfis replied:
Still applicable

AaronCrawfis commented on daprdocs/content/en/developing-applications/ides/visual-studio-code/vscode-dapr-extension.md:
```diff
@@ -0,0 +1,140 @@
+
+## Telemetry
+
+### Data Collection
```
```suggestion
### Data collection
```

AaronCrawfis commented on daprdocs/content/en/developing-applications/ides/visual-studio-code/vscode-dapr-extension.md:
```diff
@@ -0,0 +1,140 @@
+
+### Data Collection
+
+The software may collect information about you and your use of the software and send it to Microsoft. Microsoft may use this information to provide services and improve our products and services. You may turn off the telemetry as described in the repository. There are also some features in the software that may enable you and Microsoft to collect data from users of your applications. If you use t ...
```
```suggestion
The software may collect information about you and your use of the software and send it to Microsoft. Microsoft may use this information to provide services and improve our products and services. You may turn off the telemetry as described [in the repository](https://github.com/microsoft/vscode-dapr). There are also some features in the software that may enable you and Microsoft to collect data from users of your applications. If you use these features, you must comply with applicable law, including providing appropriate notices to users of your applications together with a copy of Microsoft's privacy statement. Microsoft's privacy statement is located at https://go.microsoft.com/fwlink/?LinkID=824704. You can learn more about data collection and use in the help documentation and privacy statement. Your use of the software operates as your consent to these practices.
```

AaronCrawfis commented on daprdocs/content/en/developing-applications/ides/visual-studio-code/vscode-dapr-extension.md:
```diff
@@ -0,0 +1,140 @@
+
+The software may collect information about you and your use of the software and send it to Microsoft. Microsoft may use this information to provide services and improve our products and services. You may turn off the telemetry as described in the repository. There are also some features in the software that may enable you and Microsoft to collect data from users of your applications. If you use t ...
+
+### Disabling Telemetry
+If you don’t wish to send usage data to Microsoft, you can set the telemetry.enableTelemetry setting to false. 
+
```
```suggestion
### Disabling Telemetry

If you don’t wish to send usage data to Microsoft, you can set the `telemetry.enableTelemetry` setting to `false`. Learn more in the VSCode [FAQ](https://code.visualstudio.com/docs/supporting/faq#_how-to-disable-telemetry-reporting).
```

AaronCrawfis commented on daprdocs/content/en/developing-applications/ides/visual-studio-code/vscode-dapr-extension.md:
```diff
@@ -0,0 +1,140 @@
+If you don’t wish to send usage data to Microsoft, you can set the telemetry.enableTelemetry setting to false. 
+
+
+## Additional Resources
```
```suggestion
## Additional resources
```

AaronCrawfis commented on daprdocs/content/en/developing-applications/ides/visual-studio-code/vscode-dapr-extension.md:
```diff
@@ -0,0 +1,140 @@
+
+## Additional Resources
+
+### Debuggin Multiple Dapr Applications At The Same Time
+Using the VS Code extension you can debug multiple Dapr applications at the same time with [Multi-target debugging](https://code.visualstudio.com/docs/editor/debugging#_multitarget-debugging)
+
+### Dapr Community Call Demonstration
+Watch this [video](https://www.youtube.com/watch?v=OtbYCBt9C34&t=85) on how to use the Dapr VS Code extension:
+<iframe width="560" height="315" src="https://www.youtube.com/embed/OtbYCBt9C34?start=85" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
```
```suggestion
### Debugging multiple Dapr applications at the same time

Using the VS Code extension, you can debug multiple Dapr applications at the same time with [Multi-target debugging](https://code.visualstudio.com/docs/editor/debugging#_multitarget-debugging).

### Community call demo

Watch this [video](https://www.youtube.com/watch?v=OtbYCBt9C34&t=85) on how to use the Dapr VS Code extension:

<iframe width="560" height="315" src="https://www.youtube.com/embed/OtbYCBt9C34?start=85" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
```

AaronCrawfis commented on daprdocs/content/en/developing-applications/ides/visual-studio-code/vscode-manual-configuration.md:
```diff
@@ -1,64 +1,11 @@
-weight: 1000
-description:  "Information on how to develop and run Dapr applications in VS Code"
+title: "Visual Studio Code manual configuration"
+linkTitle: "Visual Studio Code manual configuration"
```
```suggestion
linkTitle: "Manual debugging"
```

AaronCrawfis replied:
This one I think is best as "Manual debugging" as it's shorter and won't word wrap on narrow screens

AaronCrawfis commented on daprdocs/content/en/developing-applications/ides/visual-studio-code/vscode-manual-configuration.md:
```diff
@@ -1,64 +1,11 @@
+title: "Visual Studio Code manual configuration"
+linkTitle: "Visual Studio Code manual configuration"
+weight: 30000
+description:  "Information on how to setup Visual Studio Code debugging manually"
```
```suggestion
description:  "How to manually setup Visual Studio Code debugging"
```

AaronCrawfis commented on daprdocs/content/en/developing-applications/ides/visual-studio-code/vscode-manual-configuration.md:
```diff
@@ -1,64 +1,11 @@
-
-
-### Manually configuring Visual Studio Code for debugging with daprd
 If instead of using the Dapr VS Code extension you wish to configure a project to use Dapr in the [tasks.json](https://code.visualstudio.com/Docs/editor/tasks) and [launch.json](https://code.visualstudio.com/Docs/editor/debugging) files these are the manual steps.
```
```suggestion
The [Dapr VSCode extension]({{< ref vscode-dapr-extension.md >}}) automates the setup of [VSCode debugging](https://code.visualstudio.com/Docs/editor/debugging).

If instead you wish to manually configure the `[tasks.json](https://code.visualstudio.com/Docs/editor/tasks)` and `[launch.json](https://code.visualstudio.com/Docs/editor/debugging)` files to use Dapr, these are the steps.
```

AaronCrawfis commented on daprdocs/content/en/developing-applications/ides/visual-studio-code/vscode-manual-configuration.md:
```diff
@@ -1,64 +1,11 @@
-### Manually configuring Visual Studio Code for debugging with daprd
 If instead of using the Dapr VS Code extension you wish to configure a project to use Dapr in the [tasks.json](https://code.visualstudio.com/Docs/editor/tasks) and [launch.json](https://code.visualstudio.com/Docs/editor/debugging) files these are the manual steps.
 
 When developing Dapr applications, you typically use the dapr cli to start your daprized service similar to this:
```
```suggestion
When developing Dapr applications, you typically use the Dapr cli to start your daprized service similar to this:
```

AaronCrawfis commented on daprdocs/content/en/developing-applications/ides/visual-studio-code/vscode-remote-dev-containers.md:
```diff
@@ -0,0 +1,31 @@
+---
+type: docs
+title: "Visual Studio Code Remote Dev Containers"
+linkTitle: "Visual Studio Code remote dev containers"
```
```suggestion
linkTitle: "Remote dev containers"
```

orizohar replied:
Can drop the words "Visual Studio Code" here

AaronCrawfis commented on daprdocs/content/en/developing-applications/ides/visual-studio-code/vscode-remote-dev-containers.md:
```diff
@@ -0,0 +1,31 @@
+---
+type: docs
+title: "Visual Studio Code Remote Dev Containers"
```
```suggestion
title: "Visual Studio Code remote dev containers"
```

orizohar replied:
This seems like a "How To" doc. See other simialr pages. The title should reflect this is a page with instructions on how to setup VS Code with remote dev containers.

AaronCrawfis commented on daprdocs/content/en/developing-applications/ides/visual-studio-code/vscode-remote-dev-containers.md:
```diff
@@ -0,0 +1,31 @@
+title: "Visual Studio Code Remote Dev Containers"
+linkTitle: "Visual Studio Code remote dev containers"
+weight: 20000
+description:  "Information on how to setup a remote dev environment"
```
```suggestion
description:  "How to setup a remote dev container environment with Dapr"
```

AaronCrawfis commented on daprdocs/content/en/developing-applications/ides/visual-studio-code/vscode-remote-dev-containers.md:
```diff
@@ -0,0 +1,31 @@
+description:  "Information on how to setup a remote dev environment"
+---
+
+The Visual Studio Code Remote Containers extension lets you use a Docker container as a full-featured development environment enabling you to [develop inside a container](https://code.visualstudio.com/docs/remote/containers) without installing any additional frameworks or packages to your local filesystem.
```
```suggestion
The Visual Studio Code [Remote Containers extension](https://code.visualstudio.com/docs/remote/containers) lets you use a Docker container as a full-featured development environment without installing any additional frameworks or packages to your local filesystem.
```

AaronCrawfis commented on daprdocs/content/en/developing-applications/ides/visual-studio-code/vscode-remote-dev-containers.md:
```diff
@@ -0,0 +1,31 @@
+
+#### Create remote Dapr container
+1. Open your application workspace in VS Code
+2. In the command command palette (ctrl+shift+p) type and select `Remote-Containers: Add Development Container Configuration Files...`
```
```suggestion
2. In the command command palette (`CTRL+SHIFT+P`) type and select `Remote-Containers: Add Development Container Configuration Files...`
```

AaronCrawfis reviewed (changes requested):
Reorganizzation and updates look great! Added some suggests for formatting and simplicity. @orizohar the manual configuration page is out of date with Dapr CLI behavior. We'll need to update that.

orizohar commented on daprdocs/content/en/developing-applications/ides/visual-studio-code/vscode-dapr-extension.md:
```diff
@@ -0,0 +1,140 @@
+
+## Features Overview
+
+### Scaffold Dapr Tasks 
```
"Tasks" --> "tasks"

orizohar commented on daprdocs/content/en/developing-applications/ides/visual-studio-code/vscode-dapr-extension.md:
```diff
@@ -0,0 +1,140 @@
+
+* Helps scaffold VS Code task.json and launch.json configurations needed to debug your application within the Dapr environment.
+* Requires an already existing launch.json files to be found in the VS Code workspace.
+* Example NodeJS App:
```
"App" --> "app"

orizohar commented on daprdocs/content/en/developing-applications/ides/visual-studio-code/vscode-dapr-extension.md:
````diff
@@ -0,0 +1,140 @@
+}
+```
+
+### Scaffold Dapr Components 
````
"Components" --> "components"

orizohar commented on daprdocs/content/en/developing-applications/ides/visual-studio-code/vscode-dapr-extension.md:
```diff
@@ -0,0 +1,140 @@
+
+<br /><img src="/images/vscode-extension-view.png" alt="Screenshot of the Dapr VSCode extension view running applications option" width="800">
+
+### Invoke Dapr Application Methods
```
application methods

orizohar commented on daprdocs/content/en/developing-applications/ides/visual-studio-code/vscode-manual-configuration.md:
```diff
@@ -1,64 +1,11 @@
-linkTitle: "Visual Studio Code"
-weight: 1000
-description:  "Information on how to develop and run Dapr applications in VS Code"
+title: "Visual Studio Code manual configuration"
```
This page only speaks to debugging related configurations correct? if so a better title would be "... manual debugging configuration"

orizohar commented on daprdocs/content/en/developing-applications/ides/visual-studio-code/vscode-manual-configuration.md:
```diff
@@ -1,64 +1,11 @@
-weight: 1000
-description:  "Information on how to develop and run Dapr applications in VS Code"
+title: "Visual Studio Code manual configuration"
+linkTitle: "Visual Studio Code manual configuration"
```
For LinkTitle, we try to keep it short. This is already in the context of "Visual Studio Code" so can drop that from the prefix (also add "debugging" per the other comment)

orizohar reviewed (changes requested):
Looks good. See some of my minor change requests. 
I know some of this wording was there before and just refactored but please see it is also consistent with style and tone (for example, can remove the "lets get started" first person statement in the manual config page )

jasonviviano committed:
Added changes to the reformatting of the VS Code

Changes are based on comments received.

jasonviviano committed:
Revised grammer on manual configuration and remote dev containers

Revised based on comments

jasonviviano committed:
Removed unnecessary wording

jasonviviano commented:
> Looks good. See some of my minor change requests.
> I know some of this wording was there before and just refactored but please see it is also consistent with style and tone (for example, can remove the "lets get started" first person statement in the manual config page )

Got it! fixed everything except the extension title regarding "Overview"

Commit:
Merge branch 'v1.2' into jviviano

AaronCrawfis commented on daprdocs/content/en/developing-applications/ides/visual-studio-code/vscode-dapr-extension.md:
````diff
@@ -0,0 +1,136 @@
+            "request": "launch",
+            "name": "Launch Program",
+            "skipFiles": [
+                "<node_internals>/**"
+            ],
+            "program": "${workspaceFolder}/app.js"
+        },
+        {
+            "type": "pwa-node",
+            "request": "launch",
+            "name": "Launch Program with Dapr",
+            "skipFiles": [
+                "<node_internals>/**"
+            ],
+            "program": "${workspaceFolder}/app.js",
+            "preLaunchTask": "daprd-debug",
+            "postDebugTask": "daprd-down"
+        }
+}
+```
+After scaffolding task.json:
+
+```json
+{
+	"version": "2.0.0",
+	"tasks": [
+		{
+			"appId": "nodeapp",
+			"appPort": 3500,
+			"label": "daprd-debug",
+			"type": "daprd"
+		},
+		{
+			"appId": "nodeapp",
+			"label": "daprd-down",
+			"type": "daprd-down"
+		}
+	]
+}
+```
````
Way too verbose. I suggest dropping this.

AaronCrawfis replied:
See my open comments for my suggestions

jasonviviano replied:
I can just leave, "daprd-debug" for the task.json and  the "Launch Program with Dapr" content then. that way each JSON example is only like 4 to 6 lines.

jasonviviano replied:
Submitted a change to this let me know if it matches what you believe would be useful for a user.

jasonviviano committed:
Revision on JSON Snippets Shown

Cut down the JSON shown to users in the VS Code Dapr section and renamed link title

jasonviviano committed:
Merge branch 'jviviano' of https://github.com/jasonviviano/docs into jviviano

AaronCrawfis commented on daprdocs/content/en/developing-applications/ides/visual-studio-code/vscode-dapr-extension.md:
````diff
@@ -0,0 +1,96 @@
+
+## Features
+
+### Scaffold Dapr tasks 
+
+* Helps scaffold VS Code task.json and launch.json configurations needed to debug your application within the Dapr environment.
+* Requires an already existing launch.json files to be found in the VS Code workspace.
+* Example NodeJS app:
+
+After scaffolding the launch.json should contain similar snippets:
+```json
+        {
+            "type": "pwa-node",
+            "request": "launch",
+            "name": "Launch Program with Dapr",
+            "skipFiles": [
+                "<node_internals>/**"
+            ],
+            "program": "${workspaceFolder}/app.js",
+            "preLaunchTask": "daprd-debug",
+            "postDebugTask": "daprd-down"
+        }
+```
+After scaffolding the task.json should contain similar snippets:
+
+```json
+		{
+			"appId": "nodeapp",
+			"appPort": 3500,
+			"label": "daprd-debug",
+			"type": "daprd"
+		}
+```
````
```suggestion
### Scaffold Dapr debugging tasks

The Dapr extension helps you debug your applications with Dapr using Visual Studio Code's [built-in debugging capability](https://code.visualstudio.com/Docs/editor/debugging).

Using the `Dapr: Scaffold Dapr Tasks` [Command Palette](https://code.visualstudio.com/docs/getstarted/userinterface#_command-palette) operation, you can update your existing `task.json` and `launch.json` files to launch and configure the Dapr sidecar when you begin debugging.

1. Make sure you have a launch configuration set for your app. ([Learn more](https://code.visualstudio.com/Docs/editor/debugging))
1. Open the Command Palette with `Ctrl+Shift+P`
1. Select `Dapr: Scaffold Dapr Tasks`
1. Run your app and the Dapr sidecar with `F5` or via the Run view.
```

msfussell replied:
Agree with Aaron here. This is not about showing the feature this is about "how" to use the VS Core Extension feature. I certainly could not  figure out how to use the debug extension without this.

AaronCrawfis commented on daprdocs/content/en/developing-applications/ides/visual-studio-code/vscode-dapr-extension.md:
````diff
@@ -0,0 +1,96 @@
+		}
+```
+
+### Scaffold Dapr components 
+* Generates the Dapr component assets needed for general Dapr applications.
+* Provides a Component folder in your VS Code workspace that contains pubsub, statestore and a zipkin yaml file.
+* Example pubsub file:
+
+```yaml
+apiVersion: dapr.io/v1alpha1
+kind: Component
+metadata:
+  name: pubsub
+spec:
+  type: pubsub.redis
+  metadata:
+  - name: redisHost
+    value: localhost:6379
+  - name: redisPassword
+    value: ""
+```
````
```suggestion
### Scaffold Dapr components

When adding Dapr to your application, you may want to have a dedicated components directory, separate from the default components initialized as part of `dapr init`.

To create a dedicated components folder with the default `statestore`, `pubsub`, and `zipkin` components, use the `Dapr: Scaffold Dapr Components` [Command Palette](https://code.visualstudio.com/docs/getstarted/userinterface#_command-palette) operation.

1. Open your application directory in Visual Studio Code
1. Open the Command Palette with `Ctrl+Shift+P`
1. Select `Dapr: Scaffold Dapr Components`
1. Run your application with `dapr run --components-path ./components -- ...`
```

msfussell replied:
Same comment here on "How" to use the extension.

AaronCrawfis commented on daprdocs/content/en/developing-applications/ides/visual-studio-code/vscode-dapr-extension.md:
````diff
@@ -0,0 +1,96 @@
+    value: ""
+```
+
+### View running Dapr applications
+* The Applications view shows Dapr applications running locally on your machine.
+<br /><img src="/images/vscode-extension-view.png" alt="Screenshot of the Dapr VSCode extension view running applications option" width="800">
````
```suggestion
### View running Dapr applications

The Applications view shows Dapr applications running locally on your machine.

<br /><img src="/images/vscode-extension-view.png" alt="Screenshot of the Dapr VSCode extension view running applications option" width="800">
```

AaronCrawfis commented on daprdocs/content/en/developing-applications/ides/visual-studio-code/vscode-dapr-extension.md:
```diff
@@ -0,0 +1,96 @@
+* The Applications view shows Dapr applications running locally on your machine.
+<br /><img src="/images/vscode-extension-view.png" alt="Screenshot of the Dapr VSCode extension view running applications option" width="800">
+
+### Invoke Dapr application methods
+* Allows users to select a Dapr application found in the tree view "Applications" and invoke GET/POST methods by name.
+* Allows users to specify an optional payload for POST methods
+<br /><img src="/images/vscode-extension-invoke.png" alt="Screenshot of the Dapr VSCode extension invoke option" width="800">
```
```suggestion
### Invoke Dapr applications

Within the Applications view, users can right-click and invoke Dapr apps via GET or POST methods, optionally specifying a payload.

<br /><img src="/images/vscode-extension-invoke.png" alt="Screenshot of the Dapr VSCode extension invoke option" width="800">
```

AaronCrawfis commented on daprdocs/content/en/developing-applications/ides/visual-studio-code/vscode-dapr-extension.md:
```diff
@@ -0,0 +1,96 @@
+* Allows users to specify an optional payload for POST methods
+<br /><img src="/images/vscode-extension-invoke.png" alt="Screenshot of the Dapr VSCode extension invoke option" width="800">
+
+### Publish events to Dapr applications
+* Within the Applications view, users can right-click and publish messages to a running Dapr application, specifying the topic and payload.
+Users can also publish messages to all running applications.
+  <br /><img src="/images/vscode-extension-publish.png" alt="Screenshot of the Dapr VSCode extension publish option" width="800">
+
```
```suggestion
### Publish events to Dapr applications

Within the Applications view, users can right-click and publish messages to a running Dapr application, specifying the topic and payload.

Users can also publish messages to all running applications.

  <br /><img src="/images/vscode-extension-publish.png" alt="Screenshot of the Dapr VSCode extension publish option" width="800">
```

AaronCrawfis commented on daprdocs/content/en/developing-applications/ides/visual-studio-code/vscode-dapr-extension.md:
```diff
@@ -0,0 +1,96 @@
+
+## Additional resources
+
+### Debugging multiple Dapr applications at the same time
```
```suggestion
### Debugging multiple Dapr applications at the same time

```

AaronCrawfis commented on daprdocs/content/en/developing-applications/ides/visual-studio-code/vscode-dapr-extension.md:
```diff
@@ -0,0 +1,96 @@
+
+The software may collect information about you and your use of the software and send it to Microsoft. Microsoft may use this information to provide services and improve our products and services. You may turn off the telemetry as described [in the repository](https://github.com/microsoft/vscode-dapr). There are also some features in the software that may enable you and Microsoft to collect data f ...
+
+### Disabling Telemetry
```
```suggestion
### Disabling Telemetry

```

AaronCrawfis commented on daprdocs/content/en/developing-applications/ides/visual-studio-code/vscode-dapr-extension.md:
```diff
@@ -0,0 +1,96 @@
+
+### Debugging multiple Dapr applications at the same time
+Using the VS Code extension, you can debug multiple Dapr applications at the same time with [Multi-target debugging](https://code.visualstudio.com/docs/editor/debugging#_multitarget-debugging).
+### Community call demo
```
```suggestion

### Community call demo

```

AaronCrawfis commented on daprdocs/content/en/developing-applications/ides/visual-studio-code/vscode-manual-configuration.md:
```diff
@@ -1,67 +1,16 @@
-weight: 1000
-description:  "Information on how to develop and run Dapr applications in VS Code"
+title: "Visual Studio Code manual debugging configuration"
+linkTitle: "Manual debugging configuration"
```
```suggestion
linkTitle: "Manual debugging"
```

AaronCrawfis reviewed (changes requested):
Added some suggestions  on how to make the features more human readable, focusing on how to use the feature instead of a bullet on what it does. Also some small markdown style fixes

jasonviviano committed:
Update daprdocs/content/en/developing-applications/ides/visual-studio-code/vscode-dapr-extension.md

Co-authored-by: Aaron Crawfis <Aaron.Crawfis@microsoft.com>

jasonviviano committed:
Update daprdocs/content/en/developing-applications/ides/visual-studio-code/vscode-dapr-extension.md

Co-authored-by: Aaron Crawfis <Aaron.Crawfis@microsoft.com>

jasonviviano committed:
Update daprdocs/content/en/developing-applications/ides/visual-studio-code/vscode-dapr-extension.md

Co-authored-by: Aaron Crawfis <Aaron.Crawfis@microsoft.com>

jasonviviano committed:
Update daprdocs/content/en/developing-applications/ides/visual-studio-code/vscode-dapr-extension.md

Co-authored-by: Aaron Crawfis <Aaron.Crawfis@microsoft.com>

jasonviviano committed:
Update daprdocs/content/en/developing-applications/ides/visual-studio-code/vscode-dapr-extension.md

Co-authored-by: Aaron Crawfis <Aaron.Crawfis@microsoft.com>

jasonviviano committed:
Update daprdocs/content/en/developing-applications/ides/visual-studio-code/vscode-dapr-extension.md

Co-authored-by: Aaron Crawfis <Aaron.Crawfis@microsoft.com>

jasonviviano committed:
Update daprdocs/content/en/developing-applications/ides/visual-studio-code/vscode-manual-configuration.md

Co-authored-by: Aaron Crawfis <Aaron.Crawfis@microsoft.com>

jasonviviano committed:
Update daprdocs/content/en/developing-applications/ides/visual-studio-code/vscode-dapr-extension.md

Co-authored-by: Aaron Crawfis <Aaron.Crawfis@microsoft.com>

jasonviviano committed:
Update daprdocs/content/en/developing-applications/ides/visual-studio-code/vscode-dapr-extension.md

Co-authored-by: Aaron Crawfis <Aaron.Crawfis@microsoft.com>

Commit:
Merge branch 'v1.2' into jviviano

jasonviviano commented:
Hey everyone I tried to fix all the spots you all pointed out, I believe that the pages should be in good shape now @AaronCrawfis suggestions were pretty much spot on and they fixed the informal structure that the extension page had for the first two features the rest are pretty much self explanatory in their usage. Let me know if anyone has any other points i should look into @msfussell.

msfussell commented on daprdocs/content/en/developing-applications/ides/visual-studio-code/vscode-dapr-extension.md:
```diff
@@ -0,0 +1,77 @@
+
+  <br /><img src="/images/vscode-extension-publish.png" alt="Screenshot of the Dapr VSCode extension publish option" width="800">
+
+## Telemetry
```
We should not have this telemetry statement in the Dapr docs, not only because this is a VS Code issue, but because talking about Microsoft in the docs gathering data is not a good statement to have in the Dapr docs. Dapr is vendor neutral. Remove the telemetry section.

jasonviviano replied:
Sounds good! I only added it because it was unclear if we were responsible for stating that information as well as a statement of open source awareness.

msfussell commented on daprdocs/content/en/developing-applications/ides/visual-studio-code/vscode-dapr-extension.md:
```diff
@@ -0,0 +1,77 @@
+
+
+Dapr offers a *preview* [Dapr Visual Studio Code extension](https://marketplace.visualstudio.com/items?itemName=ms-azuretools.vscode-dapr) for local development and debugging of your Dapr applications.
+
```
It would be good to have a description in the intro of what the VSCode Extension is useful for, its not just debugging. 
Another thing that is not clear is can use this for any language (or at least the supported Dapr languages). I presume you can, but not sure

jasonviviano replied:
Sounds good I will change it to fully recognize its usage, and will add that it does support all languages that support Dapr as it doesn't have any language specific restrictions based on my testing and conversations with Phillip.

msfussell commented on daprdocs/content/en/developing-applications/ides/visual-studio-code/vscode-dapr-extension.md:
```diff
@@ -0,0 +1,77 @@
+1. Open the Command Palette with `Ctrl+Shift+P`
+1. Select `Dapr: Scaffold Dapr Tasks`
+1. Run your app and the Dapr sidecar with `F5` or via the Run view.
+
```
Aren;t there images (screen shots here)?

jasonviviano replied:
Not anymore, the old ones only captured a very specific step and aren't needed with a clearly stated instruction sequence that tackles the whole procedure.

msfussell commented on daprdocs/content/en/developing-applications/ides/visual-studio-code/vscode-dapr-extension.md:
```diff
@@ -0,0 +1,77 @@
+1. Open the Command Palette with `Ctrl+Shift+P`
+1. Select `Dapr: Scaffold Dapr Components`
+1. Run your application with `dapr run --components-path ./components -- ...`
+
```
Is there  a screen shots here for consistency?

jasonviviano replied:
Sure I can add that if you believe it would be nice would the screen shots be of the whole process or just the end result?

msfussell commented on daprdocs/content/en/developing-applications/ides/visual-studio-code/vscode-manual-configuration.md:
```diff
@@ -224,4 +171,4 @@ Once you have made the required changes, you should be able to switch to the [de
 
 {{% alert title="Note" color="primary" %}}
 Since you didn't launch the service(s) using the **dapr** ***run*** cli command, but instead by running **daprd**, the **dapr** ***list*** command will not show a list of apps that are currently running.
```
Read through this document and make sure this is still relevant and works. For example using the CLI to generate a components folder is wrong here (this is an old feature). It looks like this section needs updating. At the same time remove all usage of the word "will" it is better to write in the present tense, rather than the future tense.

jasonviviano replied:
Sounds good, I should've checked that!

jasonviviano replied:
Finished checking that the information is accurate, it's pretty much what the scaffold task in the extension does. It doesn't rely on Dapr instead uses daprd to to launch a service when the debug play button is used. Removed the 'will' from every spot except were it makes sense in the disclaimer at the bottom talking about the Dapr List lack of identifying daprd instances. Resolving this.

msfussell commented on daprdocs/content/en/developing-applications/ides/visual-studio-code/vscode-remote-dev-containers.md:
```diff
@@ -0,0 +1,31 @@
+
+The Visual Studio Code [Remote Containers extension](https://code.visualstudio.com/docs/remote/containers) lets you use a Docker container as a full-featured development environment without installing any additional frameworks or packages to your local filesystem.
+
+Dapr has pre-built Docker remote containers for each of the language SDKs. You can pick the one of your choice for a ready made environment. Note these pre-built containers automatically update to the latest Dapr release.
```
Is this true? I thought this was only Node and C#? Is there one for Java, Node and Python? Please confirm.

msfussell replied:
As a follow up note to this, can we check that these are still valid and pull in the latest Dapr SDK when used? Check with Philip who generate these.

jasonviviano replied:
Yeah you're right only Node and C#, I had just trusted the details that were in the current docs. And from my understanding of the code I believe so when I followed the procedures the Docker file didn't seem dapr version specific but i will ask Phillip since I should've confirmed!

msfussell commented on daprdocs/content/en/developing-applications/ides/visual-studio-code/vscode-remote-dev-containers.md:
```diff
@@ -0,0 +1,31 @@
+title: "Visual Studio Code remote dev containers"
+linkTitle: "Remote dev containers"
+weight: 20000
+description:  "How to setup a remote dev container environment with Dapr"
```
I am not a fan of this title since it does not tell you what you can do. Can we call this something like "Developing Dapr applications with remote dev containers". I have always found it confusing just saying remote dev containers, since unless you know what they are, you do not know what they are useful for.

jasonviviano replied:
Yeah you're totally right!

msfussell commented on daprdocs/content/en/developing-applications/ides/visual-studio-code/_index.md:
```diff
@@ -0,0 +1,7 @@
+---
```
Any reason the URL of this index has changed from vscode to visual-studio-code?

jasonviviano replied:
I did that since I felt the full name of VS Code would benefit more people, I know VS Code is the official alias and resonates with us easily I felt the full name was more inclusive.

orizohar replied:
Visual Studio code is the official name of the product. I think it may also help with SEO in the URL

msfussell commented on daprdocs/content/en/developing-applications/ides/visual-studio-code/_index.md:
```diff
@@ -0,0 +1,7 @@
+type: docs
+title: "Visual Studio Code integration with Dapr"
+linkTitle: "Visual Studio Code"
+weight: 1000
```
Are the new topics that are broken out sub heading under a vscode index file? I could not see them listed.

jasonviviano replied:
They are under vscode as my reasoning for the full name in the index file above. Once they were introduced to the full name I felt using the VSCode alias was more fitting now.

orizohar replied:
Hugo lists this based on the location. When the site is built these are shown up correctly

msfussell commented on daprdocs/content/en/developing-applications/ides/visual-studio-code/vscode-dapr-extension.md:
```diff
@@ -0,0 +1,77 @@
+1. Run your app and the Dapr sidecar with `F5` or via the Run view.
+
+### Scaffold Dapr components
+
```
Is this the only new feature added? I thought that there were others? Will these be in another PR and why have just this one and not say the dashboard button? Is only this one finished?

jasonviviano replied:
This was actually not a new feature but a feature that just wasn't captured in the older version as it was made by Phillip in the older versions of Dapr which didn't organize the Components folder automatically for you I believe. The newer features from 0.5 update aren't capture because I didn't want to add a feature I couldn't guarantee would be out when this document would be.

jasonviviano replied:
There are new features already implemented but they just aren't in the latest release right now.

jasonviviano committed:
Added more details based on Mark's comments

jasonviviano committed:
Fixed vscode link and updated information on manual.

Commit:
Merge branch 'v1.2' into jviviano
